### PR TITLE
[ONEM-34706] Refactor rdkservices WebkitBrowser plugin code related to setting up client certificates required for mTLS

### PR DESCRIPTION
**Pull request by franciszek-koltuniuk-red**
Implemented a new flow to process the certificate data signaled in at runtime as a request query parameters &clientcert=<client data> and &clientcertkey=<client data>

### CODE REVIEW

**Review by jacek-manko-red — DISMISSED** [review]

**Review by mikolaj-staworzynski-red — DISMISSED** [review]

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+                {
+                    if (url == newWpeHost)
+                    {
+                        TRACE(Trace::Information, (_T("Already found on the list %s\n "), newWpeHost.c_str()));
+                        append_newHost = false;
+                        break;
+                    }
+                }
+
+                if (append_newHost)
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp:2011]: I would append the host only if all the operations succeeded (certContents and keyContents are fine).

File: WebKitBrowser/WebKitImplementation.cpp [hunk 2]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+                        urlsList += kUrlsSep;
+                    }
+                    urlsList += newWpeHost;
+                }
+
+                std::string certContents = GetFileContent(newWpeClientCert);
+
+                if (certContents.empty())
+                {
+                    TRACE(Trace::Error, (_T("Empty certificate for %s %s"), newWpeClientCert.c_str(), newWpeHost.c_str()));
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp:2024]: Break and not update ENV_WPE_CLIENT_CERTIFICATES_URLS.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 3]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+                }
+                std::string keyContents = GetFileContent(newWpeClientCertKey);
+                if (keyContents.empty())
+                {
+                    TRACE(Trace::Error, (_T("Empty private key for %s %s"), newWpeClientCertKey.c_str(), newWpeHost.c_str()));
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: Break and not update ENV_WPE_CLIENT_CERTIFICATES_URLS.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 4]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+                }
+                if (!DecryptWithOpenSSL(&keyContents))
+                {
+                    TRACE(Trace::Error, (_T("Failed decrypting private key for %s %s"), newWpeClientCertKey.c_str(), newWpeHost.c_str()));
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: Break and not update ENV_WPE_CLIENT_CERTIFICATES_URLS.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 5]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+                                new_url.append("?");
+                            }
+                        }
+                        new_url.append(urlKeyStr);
+                    }
+                }
+                if(new_host.size() && new_payload_cert.size() && new_payload_certkey.size() ) {
+                    AppendClientCertificate(new_host, new_payload_cert, new_payload_certkey, newCertContents);
+                }
+                TRACE_L1("New URL: %s (Base URL: %s)", new_host.c_str(), url.c_str());
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: I think it's the other way around.
New URL: is full url (url.c_str())
Base URL: is origin url (new_host.c_str())

File: WebKitBrowser/WebKitImplementation.cpp [hunk 6]
```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
...
+
+            } else {
+                new_url = string(url);
+            }
+            return string(new_url);
+        }
+
+        uint32_t URL(const string& URLNEW) override
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: Can we change NEWURL to fullURL or URLwithParams or something like that?
fullURL will indicate that it contains certs, and URL will be stripped fullURL without those additional parameters.

**Review by marcin-mielczarczyk-red — DISMISSED** [review]

File: WebKitBrowser/WebKitImplementation.cpp (hunk 6 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp]: updated to: URLwithParams

File: WebKitBrowser/WebKitImplementation.cpp (hunk 5 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp]: Updated to:
 New URL: is a full new URL ( new_url.c_str())
 Base URL: is a full origin URL (url.c_str())

File: WebKitBrowser/WebKitImplementation.cpp (hunk 4 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp]: Done

File: WebKitBrowser/WebKitImplementation.cpp (hunk 3 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp]: Done

File: WebKitBrowser/WebKitImplementation.cpp (hunk 2 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp:2024]: Done

File: WebKitBrowser/WebKitImplementation.cpp (hunk 1 above)
Comment (franciszek-koltuniuk-red) [reply WebKitBrowser/WebKitImplementation.cpp:2011]: Done

**Review by mikolaj-staworzynski-red — APPROVED** [review]

**Review by marcin-mielczarczyk-red — APPROVED** [review]

**Pull request merged**